src/files/file-import-status/FileImportStatus.js

```diff
@@ -13,37 +13,75 @@ import GlyphTick from '../../icons/GlyphTick'
 import GlyphCancel from '../../icons/GlyphCancel'
 import GlyphSmallCancel from '../../icons/GlyphSmallCancel'
 
-const File = (job, t) => {
-  const pathsByFolder = job.message.entries.reduce((prev, currentEntry) => {
-    const isFolder = currentEntry.path.includes('/')
-    if (!isFolder) {
-      return [...prev, currentEntry]
-    }
-
-    const baseFolder = currentEntry.path.split('/')[0]
-
-    const alreadyExistentBaseFolder = prev.find(previousPath => previousPath.path.startsWith(`${baseFolder}/`))
+const Import = (job, t) =>
+  [...groupByPath(job.message.entries).values()].map(item => (
+    <li className="flex w-100 bb b--light-gray items-center f6 charcoal" key={item.path}>
+      {viewIcon(item)}
+      <span className="fileImportStatusName truncate">{item.path}</span>
+      <span className='gray mh2'> |
+        { item.entries && (<span> { t('filesImportStatus.count', { count: item.entries.length }) } | </span>) }
+        <span className='ml2'>{ filesize(item.size) }</span>
+      </span>
+      {viewImportStatus(job)}
+    </li>
+  ))
 
-    if (alreadyExistentBaseFolder) {
-      alreadyExistentBaseFolder.count = alreadyExistentBaseFolder.count + 1
+const viewIcon = (entry) => entry.type === 'directory'
+  ? <FolderIcon className='fileImportStatusIcon fill-aqua pa1'/>
+  : <DocumentIcon className='fileImportStatusIcon fill-aqua pa1'/>
 
-      return prev
+const viewImportStatus = (job) => {
+  switch (job.status) {
+    case 'Pending': {
+      return (<LoadingIndicator />)
+    }
+    case 'Failed': {
+      return (<GlyphCancel className="dark-red w2 ph1" fill="currentColor"/>)
+    }
+    default: {
+      return (<LoadingIndicator complete={true}/>)
     }
+  }
+}
 
-    return [...prev, { ...currentEntry, name: baseFolder, count: 1 }]
-  }, [])
+const groupByPath = (entries) => {
+  const groupedEntries = new Map()
+  for (const entry of entries) {
+    const name = baseDirectoryOf(entry)
+    if (name == null) {
+      groupedEntries.set(entry.path, { type: 'file', ...entry })
+    } else {
+      // add `/` to avoid collision with file names.
+      const directory = groupedEntries.get(`${name}/`)
+      if (directory) {
+        directory.entries.push(entry)
+        directory.size += entry.size
```

Comment:
❓ Wouldn't it be best to not mutate the Map item but instead set it again? Maybe something like

```javascript
if (directory) {
    groupedEntries.set(`${name}/`, {
          type: 'directory',
          size: directory.size + entry.size,
          path: name,
          entries: [...directory.entries, entry]
        })
}
```